app/controllers/proposal/collection_draft_proposals_controller.rb

```diff
@@ -62,11 +67,47 @@ def destroy
       end
     end
 
+    def progress
+      authorize get_resource
+
+      @status_history = get_resource.status_history || {}
+      approver_feedback = get_resource.approver_feedback || {}
+
+      if get_resource.in_work?
+        @first_stage = 'In Work'
+
+        # This applies the same validation as is applied on the preview/show page
+        show_view_setup
```

Comment:
I think it is inconsistent to make submit the only action that cannot be done on this page, so I think this validation is necessary.  I extracted it into a helper function in the original class.